### PR TITLE
Add default ChancedOutputs IRecipeComponent

### PR DESCRIPTION
**Pull request by Witixin1512**

### CODE REVIEW

File: Common/src/main/java/com/blamejared/crafttweaker/api/recipe/component/BuiltinRecipeComponents.java [hunk 1]
```diff
@@ -139,6 +142,12 @@ public static final class Output {
                 RecipeComponentEqualityCheckers::areStacksEqual
         );
         
+        public static final IRecipeComponent<List<Percentaged<IItemStack>>> CHANCED_OUTPUTS = IRecipeComponent.simple(
+            CraftTweakerConstants.rl("output/chanced_outputs"),
```
Comment (jaredlll08) [inline Common/src/main/java/com/blamejared/crafttweaker/api/recipe/component/BuiltinRecipeComponents.java]: Kinda weird to specify output twice
```suggestion
            CraftTweakerConstants.rl("output/chanced_items"),
```

File: Common/src/main/java/com/blamejared/crafttweaker/api/recipe/component/RecipeComponentEqualityCheckers.java [hunk 2]
```diff
@@ -29,6 +31,11 @@ public static <T extends Number> boolean areNumbersEqual(final T a, final T b) {
         return a.doubleValue() == b.doubleValue();
     }
     
+    public static boolean areChanceStacksEqual(final List<Percentaged<IItemStack>> a, final List<Percentaged<IItemStack>> b) {
+    
+        return a.equals(b);
+    }
+
```
Comment (jaredlll08) [inline Common/src/main/java/com/blamejared/crafttweaker/api/recipe/component/RecipeComponentEqualityCheckers.java]: Could this not just be `Objects::equals`?

**Review by jaredlll08 — CHANGES_REQUESTED** [review]

File: Common/src/main/java/com/blamejared/crafttweaker/api/recipe/component/RecipeComponentEqualityCheckers.java [hunk 3]
```diff
@@ -2,7 +2,9 @@
 
 import com.blamejared.crafttweaker.api.ingredient.IIngredient;
 import com.blamejared.crafttweaker.api.item.IItemStack;
+import com.blamejared.crafttweaker.api.util.random.Percentaged;
 
+import java.util.List;
```
Comment (jaredlll08) [inline Common/src/main/java/com/blamejared/crafttweaker/api/recipe/component/RecipeComponentEqualityCheckers.java]: These imports are now useless

**Review by jaredlll08 — APPROVED** [review]

File: Common/src/main/java/com/blamejared/crafttweaker/api/recipe/component/RecipeComponentEqualityCheckers.java (hunk 3 above)
Comment (Witixin1512) [reply Common/src/main/java/com/blamejared/crafttweaker/api/recipe/component/RecipeComponentEqualityCheckers.java:7]: image
I disagree

**Comment by TheSilkMiner** [comment]
Why is this component a single component even if it is clearly saying it's composite by being `List<T>`?

**Pull request merged**